src/main/java/edu/harvard/iq/dataverse/api/DownloadInstanceWriter.java

```diff
@@ -302,8 +303,8 @@ public void writeTo(DownloadInstance di, Class<?> clazz, Type type, Annotation[]
                     // Provide both the "Content-disposition" and "Content-Type" headers,
                     // to satisfy the widest selection of browsers out there. 
                     
-                    httpHeaders.add("Content-disposition", "attachment; filename=\"" + fileName + "\"");
-                    httpHeaders.add("Content-Type", mimeType + "; name=\"" + fileName + "\"");
+                    httpHeaders.add("Content-disposition", "attachment; filename=\"" + URLEncoder.encode(fileName, "UTF-8") + "\"");
```

Comment:
FWIW: comments in https://github.com/IQSS/dataverse/pull/4542/files  re S3 downloads suggests that Firefox doesn't/didn't like this form and needed filename* and an explicit UTF-8. Has that been checked? 

Also - the S3 version includes a fix for spaces being encoded as '+' signs by the addition of the URLEncoder.encode method. That would be a change to this same line (and the next one? I'm not sure that S3 has the equivalent (adding a name to the Content-Type header), so I'm not sure.)

Reply:
I've been testing with Firefox the whole time so I'm not worried about that browser.

Good catch on how the encoding I added changes spaces to plus signs. I just pushed 702b505 to make sure they stay as spaces on download.